### PR TITLE
Allow SanitizeBoundingBoxes to sanitize more labels

### PR DESCRIPTION
**Pull request by NicolasHug**
Closes https://github.com/pytorch/vision/issues/8294

We allow the `labels_getter` parameter of `SanitizeBoudingBoxes` to return a tuple. Each entry in the tuple gets sanitized/subsampled, along with the boxes. This enables the sanitization of properties like "area" or "iscrowd" from COCO.

cc @vfdev-5

### CODE REVIEW

File: test/test_transforms_v2.py [hunk 1]
```diff
@@ -5706,7 +5706,16 @@ def _get_boxes_and_valid_mask(self, H=256, W=128, min_size=10):
         return boxes, expected_valid_mask
 
     @pytest.mark.parametrize("min_size", (1, 10))
-    @pytest.mark.parametrize("labels_getter", ("default", lambda inputs: inputs["labels"], None, lambda inputs: None))
+    @pytest.mark.parametrize(
+        "labels_getter",
+        (
+            "default",
+            lambda inputs: inputs["labels"],
+            lambda inputs: (inputs["labels"], inputs["other_labels"]),
```
Comment (NicolasHug) [inline test/test_transforms_v2.py:5714]: This parametrization is not changed, I just added that one line to test for tuple outputs

Comment (pmeier) [reply test/test_transforms_v2.py:5714]: Should we also add one for lists given that we allow that?

File: torchvision/transforms/v2/_misc.py [hunk 2]
```diff
@@ -356,18 +365,29 @@ def forward(self, *inputs: Any) -> Any:
...
-            )
+        if labels is not None:
+            msg = "The labels in the input to forward() must be a tensor or None, got {type} instead."
+            if isinstance(labels, torch.Tensor):
+                labels = (labels,)
+            elif isinstance(labels, (tuple, list)):
+                labels = tuple(labels)
+                for entry in labels:
+                    if not isinstance(entry, torch.Tensor):
+                        # TODO: we don't need to enforce tensors, just that entries are indexable as t[bool_mask]
```
Comment (pmeier) [inline torchvision/transforms/v2/_misc.py:376]: Just checking: was this a requirement from the issue or a nice to have?

File: torchvision/transforms/v2/_misc.py [hunk 3]
```diff
@@ -356,18 +365,29 @@ def forward(self, *inputs: Any) -> Any:
...
-        if labels is not None and not isinstance(labels, torch.Tensor):
-            raise ValueError(
-                f"The labels in the input to forward() must be a tensor or None, got {type(labels)} instead."
-            )
+        if labels is not None:
+            msg = "The labels in the input to forward() must be a tensor or None, got {type} instead."
+            if isinstance(labels, torch.Tensor):
+                labels = (labels,)
+            elif isinstance(labels, (tuple, list)):
+                labels = tuple(labels)
```
Comment (pmeier) [inline torchvision/transforms/v2/_misc.py:373]: Why force into `tuple` here when we have already established that it can only be a tuple or list above?

**Review by pmeier — APPROVED** [review]
Just minor things. Thanks Nicolas!

File: torchvision/transforms/v2/_misc.py (hunk 2 above)
Comment (NicolasHug) [reply torchvision/transforms/v2/_misc.py:376]: Just a nice to have

**Comment by MarcSzafraniec** [comment]
Thanks ! :)

**Pull request merged**